src/domain/orders/details/address-modal.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect } from "react"
+import { useForm } from "react-hook-form"
+import Button from "../../../components/fundamentals/button"
+import Input from "../../../components/molecules/input"
+import Modal from "../../../components/molecules/modal"
+
+type AddressModalProps = {
+  handleClose: () => void
+  handleSave: ({ data, type }) => Promise<void>
+  address?: object
+  email?: string
+  type: "shipping" | "billing"
+}
+
+const AddressModal: React.FC<AddressModalProps> = ({
+  address,
+  email,
+  handleClose,
+  handleSave,
+  type,
+}) => {
+  const { register, handleSubmit, reset } = useForm()
+
+  useEffect(() => {
+    reset({
+      ...address,
+      email,
+    })
+  }, [])
```

Comment:
nit: i think you can pass default values to `useForm` like this `useForm({ defaultValues: { address, email } })`